clients/python/src/model_registry/_client.py

```diff
@@ -45,47 +73,57 @@ def __init__(
         Keyword Args:
             author: Name of the author.
             is_secure: Whether to use a secure connection. Defaults to True.
-            user_token: The PEM-encoded user token as a string. Defaults to content of path on envvar KF_PIPELINES_SA_TOKEN_PATH.
-            custom_ca: Path to the PEM-encoded root certificates as a string. Defaults to path on envvar CERT.
+            user_token: The PEM-encoded user token as a string.
+            user_token_envvar: Environment variable to read the user token from if it's not passed as an arg. Defaults to KF_PIPELINES_SA_TOKEN_PATH.
+            custom_ca: Path to the PEM-encoded root certificates as a string.
+            custom_ca_envvar: Environment variable to read the custom CA from if it's not passed as an arg.
+            log_level: Log level. Defaults to logging.WARNING.
         """
+        logger.setLevel(log_level)
+
         import nest_asyncio
 
+        logger.debug("Setting up reentrant async event loop")
         nest_asyncio.apply()
 
         # TODO: get remaining args from env
         self._author = author
 
-        if not user_token:
+        if not user_token and user_token_envvar:
+            logger.info("Reading user token from %s", user_token_envvar)
+            if user_token_envvar == DEFAULT_USER_TOKEN_ENVVAR:
+                logger.warning(
+                    "Using default user token envvar, if you want to use a different one, pass it as `user_token`"
+                )
             # /var/run/secrets/kubernetes.io/serviceaccount/token
-            sa_token = os.environ.get("KF_PIPELINES_SA_TOKEN_PATH")
+            sa_token = os.environ.get(user_token_envvar)
             if sa_token:
                 user_token = Path(sa_token).read_text()
             else:
                 warn("User access token is missing", stacklevel=2)
 
         if is_secure:
-            root_ca = None
-            if not custom_ca:
-                if cert := os.getenv("CERT"):
-                    root_ca = cert
+            if not custom_ca and custom_ca_envvar:
+                logger.info(
+                    "Custom CA not provided, checking for envvar %s",
+                    custom_ca_envvar,
+                )
```

Comment:
Analogous. I would move this below, because in the most cases as custom_ca is not provided (relying on the system defined one) and custom_ca_envvar empty None, this would print out.

I would move this into line 112 saying that we assume custom_ca_envvar contains the intended CA.